src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -170,21 +170,23 @@ public bool HasResponseStarted
             get { return _requestProcessingStatus == RequestProcessingStatus.ResponseStarted; }
         }
 
-        protected FrameRequestHeaders FrameRequestHeaders => _frameHeaders.RequestHeaders;
+        protected FrameRequestHeaders FrameRequestHeaders { get; private set; }
+
+        protected FrameResponseHeaders FrameResponseHeaders { get; private set; }
 
         public void InitializeHeaders()
         {
-            if (_frameHeaders == null)
+            if (FrameRequestHeaders == null)
             {
-                _frameHeaders = new Headers(ServerOptions);
-                RequestHeaders = _frameHeaders.RequestHeaders;
-                ResponseHeaders = _frameHeaders.ResponseHeaders;
+                RequestHeaders = FrameRequestHeaders = new FrameRequestHeaders();
```

Comment:
@halter73 Uuuh. Isn't [this](https://github.com/khellang/KestrelHttpServer/blob/4f036a04353640f60703370dea45748d600d2b30/src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs#L264) resetting `RequestHeaders`? `FrameRequestHeaders` and `RequestHeaders` are the same, [remember](https://github.com/khellang/KestrelHttpServer/blob/4f036a04353640f60703370dea45748d600d2b30/src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs#L221)? :wink: Which in turn [is the same as](https://github.com/khellang/KestrelHttpServer/blob/4f036a04353640f60703370dea45748d600d2b30/src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.FeatureCollection.cs#L159) `IHttpRequestFeature.Headers`...